nav2_behavior_tree/plugins/condition/goal_reached_condition.cpp

```diff
@@ -62,6 +62,9 @@ class GoalReachedCondition : public BT::ConditionNode
     node_->get_parameter_or<double>("goal_reached_tol", goal_reached_tol_, 0.25);
     tf_ = config().blackboard->get<std::shared_ptr<tf2_ros::Buffer>>("tf_buffer");
 
+    node_->declare_parameter("transform_tolerance", rclcpp::ParameterValue(0.1));
```

Comment:
This should check declare if not already declared (I think there's a nav2_util for that) since this is in the BT and you could have multiple instances of this same plugin and all will try to declare. I think declaring multiple times will cause it to crash.

Reply:
You should declare this in the BT navigator, like you did with the recover server, and then each plugin just gets it. 

I also could see this plugin just "not" having a specific parameters and using the defaults. This is just looking if we're in a threshold, so I don't think there's a need for a unique timeout since it just needs something without any particularly close synchronization.